File: app/src/main/java/com/OverSadBoy/samurairise/dagger/PresenterModule.java
package com.OverSadBoy.samurairise.dagger;

import com.OverSadBoy.samurairise.model.Model;
import com.OverSadBoy.samurairise.presenter.Presenter;

import dagger.Module;
import dagger.Provides;

@Module
public class PresenterModule {
    @Provides
    Presenter providePresenter(Model model) {
        return new Presenter(model);
    }
}

File: app/src/main/java/com/OverSadBoy/samurairise/dagger/PresenterModule.kt
package com.OverSadBoy.samurairise.dagger

import androidx.test.runner.AndroidJUnit4
import com.OverSadBoy.samurairise.model.Model
import com.OverSadBoy.samurairise.presenter.Presenter
import dagger.Module
import dagger.Provides
import org.junit.runner.RunWith

@Module
class PresenterModule {
    @Provides
    fun providePresenter(model: Model?): Presenter? {
        return Presenter(model)
    }
}